### ISSUE TITLE
Create with OverriddenCreatedOn set does not override CreatedOn

### ISSUE THREAD

**Issue by HenricF** [jordimontana82/fake-xrm-easy#211]
From http://mostlymscrm.blogspot.ca/2012/07/understanding-createdon-and.html:

> If a custom date is to be stored in the createdon field, we need to instead set the overriddencreatedon field to the desired date. Once the overriddencreatedon field is set, CRM swaps the ‘real’ date of creation i.e. today (July 17, 2012) with overriddencreatedon and sets createdon as per our requirement. That's it and you are done. But (there always is a 'but'!) this is valid ONLY for the CRM create operation.

I need to filter invoices based on their CreatedOn being > X days ago, in my workflow. I would like to set up tests with invoices created more than/less than X days ago, and confirm that they're filtered correctly.

However, the CreatedOn field cannot be set directly ("Property or indexer 'Invoice.CreatedOn' cannot be assigned to -- it is read only"). Hence, I thought I could set the OverriddenCreatedOn field, based on the description above. However, it does not seem to override the CreatedOn field.

**Steps to reproduce:**

1. Setup a context and create an invoice with the OverriddenCreatedOn field set

            var invoice = new Invoice
            {
                OverriddenCreatedOn = new DateTime(2014, 1, 1)
            };

            var fakedContext = new XrmFakedContext();

            fakedContext.Initialize();

            var service = fakedContext.GetFakedOrganizationService();
            service.Create(invoice);
2. Retrieve the created invoice.
            
            var createdInvoice = fakedContext.CreateQuery<Invoice>()
                .First();
3. Inspect the fields on the created invoice.

**Expected result:**

            createdInvoice.CreatedOn = 2014-01-01

            createdInvoice.OverriddenCreatedOn = DateTime.Now

**Actual result:**

            createdInvoice.CreatedOn = DateTime.Now

            createdInvoice.OverriddenCreatedOn = 2014-01-01

**Question**
Is it possible to make FakeXrmEasy behave as CRM, i.e. swap OverriddenCreatedOn with CreatedOn upon creation (if OverriddenCreatedOn contains data)? Or is this functionality already built into FakeXrmEasy, but I'm not creating the entities correctly?

I took a quick look at the process of creating entities in FakeXrmEasy, but I'm not familiar enough with FakeItEasy to see if/where this could be implemented.

**Comment by eNeRGy164** [comment]
If you want to initialize data, this can be done without the `service.Create` method:

```csharp
var context = new XrmFakedContext();

var invoice = new Invoice
{
    Id = Guid.NewGuid(),
    ["createdon"] = new DateTime(2013, 12, 31, 23, 0, 0, DateTimeKind.Utc)
}
entities.Add(invoice);

context.Initialize(entities);
```

**Comment by HenricF** [comment]
Thank you @eNeRGy164! This solves my problems for now. However, I'm unsure if I should close this issue, as having OverriddenCreatedOn/CreatedOn behave as in CRM might be useful in other cases.

**Comment by jordimontana82** [comment]
Thanks guys, indeed it can be injected via properties, but I also think the fake create executor should also honour the OverridenCreatedOn

**Comment by jordimontana82** [comment]
This is now available in version 1.32.1 along with the latest merged pull requests.

**Comment by HenricF** [comment]
Awesome @jordimontana82, thanks!
